### ISSUE TITLE
Group transactions by user session

### ISSUE THREAD

**Issue by cristiam86** [yeagerai/genlayer-studio#598]
We need to add a way to group transactions for every local net user.
This is important for the hosted version of the simulator, so every backend only works with its transactions.
For this purpose, we will generate a session key in the Studio frontend, store it in LocalStorage, and set it on the backend. Every created transaction should have that key, and the backend must only query transactions ensuring that Key value.

This is only for management and shouldn't interfere with the GenLayer Protocol simulation.

**Comment by cristiam86** [comment]
@AgustinRamiroDiaz @denishacquin do you think that the implementation we made for the logs can be extended/refactor for this purpose?

The mean reason to do this is to prevent users crashing other users localnetworks. We are going to deploy a backend container per user using the localnet.

**Comment by denishacquin** [comment]
This is exactly what I was mentioning in my comments on the #583 PR before I added the subscription system.

Isn't this mitigated by the fact that users subscribe to transactions explicitly from the frontend however?

The way it goes in the above-mentioned PR:

- User initiates a TX from the frontend (deploy/write) and gets hash X.
- User sends a request to subscribe to all events with hash X as topic.
- On said request, backend adds the user to a room with the hash X as topic.
- When a TX status updates, backend dispatches socket events to all users who are in the room with transaction hash X as topic.
- Only the user who initiated the TX gets the socket event, and its UI gets updated.

This new system does not impact the other socket events, who are still filtered based on the socket client ID (we introduced that filtering when we first hosted the simulator to avoid sending all events to all users). That socket client ID gets generated on each page load and gets sent along with all rpc requests and gets used to filter subsequent events.

Now, I am not sure I understand how a user could "crash another user's local network" since each user has their own docker image?
